runtime/src/main/java/org/corfudb/runtime/view/ManagementView.java

```diff
@@ -21,64 +24,77 @@
 @Slf4j
 public class ManagementView extends AbstractView {
 
-    private final long layoutRefreshTimeout = 500;
+    static final Duration WORKFLOW_TIMEOUT = Duration.ofMinutes(10);
+    static final Duration QUERY_INTERVAL = Duration.ofSeconds(1);
+    static final int WORKFLOW_RETRY = 3;
 
     public ManagementView(@NonNull CorfuRuntime runtime) {
         super(runtime);
     }
 
-    /**
-     * Checks if a given workflow with the given workflow ID is active or not.
-     * If returns True - the workflow is active
-     * If returns False - the workflow is either completed successfully or failed.
-     *
-     * @param server     Node endpoint on which the workflow is running.
-     * @param workflowId Workflow UUID
-     * @return True if active. Else False
-     */
-    public boolean isWorkflowActive(String server, UUID workflowId) {
-        return layoutHelper(l -> {
-            QueryResponse response = runtime.getRouter(server)
-                    .getClient(ManagementClient.class)
-                    .queryRequest(workflowId);
-            log.info("isWorkflowActive: {} on server:{} = {}",
-                workflowId, server, response.isActive());
-            return response.isActive();
-        });
+    void waitForWorkflow(UUID workflow, ManagementClient client) {
+        long tries = WORKFLOW_TIMEOUT.getSeconds() / QUERY_INTERVAL.getSeconds();
+        log.info("waitForWorkflow: num tries {}", tries);
+        for (long x = 0; x < tries; x++) {
+            try {
+                if (!client.queryRequest(workflow).isActive()) {
+                    break;
+                }
+            } catch (NetworkException e) {
+                log.warn("waitForWorkflow: Network exception while waiting for {} on try {}", workflow, x);
+            }
+            Sleep.sleepUninterruptibly(QUERY_INTERVAL);
+            log.info("waitForWorkflow: waiting for {} on try {}", workflow, x);
+        }
```

Comment:
Seems like it should throw a TimeoutException here?

Reply:
Throw a TimeoutException if the workflow takes more than the max wait time?

Reply:
or retries?

Reply:
It makes sense to throw a TimeoutException when waitForWorkflow executes without the "break" ever being called, I don't see why you would throw a TimeoutException on retrying within the same method.

Reply:
That's what I meant. Throw an exception if the workflow is still active after X retries, don't exit silently.

Reply:
Done.